### ISSUE TITLE
[Bug Report] Failing to build `e7cd350` with debug enabled due to variable ordering in `command_queue_interface.hpp`

### ISSUE THREAD

**Issue by marty1885** [tenstorrent/tt-metal#15356]
**Describe the bug**

The latest commit, `e7cd350de921a4aa3d5ee2b2a75ee34f9fb48f2a` fails to build on my machine under `RelWithDebugInfo` due to a bug in `command_queue_interface.hpp`. The variable `l1_alignment` is used in an assert statement before deceleration. CI likely didn't catch it due to TT_ASSERT is not compiled in a release build.

https://github.com/tenstorrent/tt-metal/blob/e7cd350de921a4aa3d5ee2b2a75ee34f9fb48f2a/tt_metal/impl/dispatch/command_queue_interface.hpp#L179-L191

To be specific

```c++
// l1_alignment used there
TT_ASSERT(DISPATCH_MESSAGE_ENTRIES <= DISPATCH_MESSAGES_MAX_OFFSET / l1_alignment + 1, "Number of dispatch message entries exceeds max representable offset");

uint32_t pcie_alignment = tt::tt_metal::hal.get_alignment(tt::tt_metal::HalMemType::HOST);
uint32_t l1_alignment = tt::tt_metal::hal.get_alignment(tt::tt_metal::HalMemType::L1); // But declared here
uint8_t num_dev_cq_addrs = magic_enum::enum_count<CommandQueueDeviceAddrType>();
```

**To Reproduce**
1. Grab the latest source code
2. In the build directory run `cmake .. -DCMAKE_BUILD_TYPE=RelWithDebugInfo`
3. Build and see error

**Expected behavior**
Should build without problem under all build mode

**Screenshots**

**Please complete the following environment information:**
- Commit: e7cd350de921a4aa3d5ee2b2a75ee34f9fb48f2a

**Additional context**

**Comment by tt-aho** [comment]
Hey Marty, thanks for catching. My mistake, posting a fix soon.

**Comment by marty1885** [comment]
Thanks for the super quick response!

BTW, for these really minor fixes, do you prefer me posting the issue only? Or make a quick PR to get it fixed?
I stopped posting PRs as I don't have write access, so merging is slow, painful and involves your people helping. But if it's easier (mentally, or something else) for you guys, I totally have no problem submitting PRs.

**Comment by ayerofieiev-tt** [comment]
It’s sure helps a lot when a fix is ready. But it is also for sure not as practical as we’d like, because we can’t run CI for a branch from a fork..

Approach case by case. No guidance for now. 
The team is aware that this constraint is critical to address or at least somehow ease the pain.

**Comment by tt-aho** [comment]
I've merged the fix to main [06c2663](https://github.com/tenstorrent/tt-metal/commit/06c2663f61cea757276f225ec5c922ed67f7d3ce), and verified building with `RelWithDebugInfo` works locally for me now. Let us know if there are still any issues.
